traits/tests/test_trait_list_dict.py

```diff
@@ -115,3 +115,27 @@ def test_pickle_whole():
     loads(dumps(a))
     b = B(dict=dict(a=a))
     loads(dumps(b))
+
+
+def test_trait_set_object_operations():
+    a = A()
+    a.aset.update({10: 'a'})
+    assert a.aset == set([0, 1, 2, 3, 4, 10])
```

Comment:
Because that was the issue that I was fixing from #289 - the various methods work correctly for sets, but not other collections.